### ISSUE TITLE
Encoding management when submitting a job (for site not un EBCDIC IBM-037)

### ISSUE THREAD

**Issue by FALLAI-Denis** [zowe/zowe-cli#876]
A Job contains technical information such as the JCL, but can also include Data in inline sysins.

This Data is sensitive to the encoding used on the z/OS mainframe system. When sending the Job, this Data must be correctly transcoded.

Use case:
- locally the Job is encoded in UTF-8
- on the z/OS mainframe system, we want the Job, and Data in sysins, to be encoded in EBCDIC IBM-1147 (France)

However, encoding management is not provided for in the processing of Jobs by Zowe CLI (and probably z/OSMF) . It seems to us that the transcoding is done from UTF-8 to EBCDIC IBM-037 (US) or (EBCDIC IBM-1047 ?), which does not suit us.

The same problem arises for the retrieval of reports and outputs produced by the execution of a Job. In our case we must apply a transformation from EBCDIC IBM-1147 to UTF-8 and not from EBCDIC IBM-037 (or EBCDIC IBM-1047) to UTF-8.

**Comment by FALLAI-Denis** [comment]
Hi,

About X-IBM-Intrdr-Mode header of submit job z/OSMF API:

```
X-IBM-Intrdr-File-Encoding
This optional header specifies that the EBCDIC code page is to be used for encoding the data that is written to the internal reader.
If not specified, the default is IBM-1047. 
This header is ignored when optional header X-IBM-Intrdr-Mode is set to RECORD or BINARY.
This header is effective only when the JCL for the job to be submitted is included in the HTTP request body.
This header is ignored when the job to be submitted resides in a data set or UNIX file.
```

For spool retrieval, there is no equivalent header. So spool file must be retrieve with a `mode=text;fileEncoding=code-page` query parameter on the URI. 

```
mode=text;fileEncoding=code-page
Specifies an alternative code page (EBCDIC) for the spool file on z/OS; the encoded text is converted to the client's request encoding. 
If not specified, the default code page is IBM-1047. 
```

**Comment by awharn** [comment]
This change has been staged, and will be available in version 7.25.0 with the `--job-encoding IBM-1147` option on the `zowe jobs submit local-file` and `zowe jobs submit stdin` commands. For downloading spool files, we also now accept the `--encoding` option on the download commands.
